### ISSUE TITLE
i18n locale overrides other locales

### ISSUE THREAD

**Issue by ivo-ljubicic** [strapi/strapi#20409]
### Required System information

- Node.js version: 18.19.0
- NPM version: 10.5.1
- Strapi version: 4.24.3
- Database: postgres
- Operating system: Mac OS darwin-x64
- Is your project Javascript or Typescript: typescript

### Describe the bug
Hi all,
I have started creating a new app and opted to use Strapi, however I have an issue with i18n.

The issue is that when I try to change internationalization I manage to select one in the dropdown but fields (which do have an internationalization enabled icon) are prefilled with data from the old locale. when i change and save something the original locale changes as well and i don’t have a different version of the locale in my db.
Also when i click fill with another localization I have nothing to choose from.

Some other fields work as expected and change the localization normally.

### Steps to reproduce the behavior

It's hard to reproduce since i've come across it only on two fields (one collection and one single type).
create a type (internationalization true) go to content manager create a collection and sometimes that collection has options in one dropdown so change language but whenever i change something it mirrors to the other language.

### Expected behavior
When internationalization changes present me with another language version or empty if it wasn't filled previously. Fill with another language should offer me another language but it doesn't.

### Screenshots

![Screenshot 2024-05-31 at 11 56 08](https://github.com/strapi/strapi/assets/166009197/fad321ce-2ba6-4dbb-a372-5166c3c90ccf)
![Screenshot 2024-05-31 at 11 56 19](https://github.com/strapi/strapi/assets/166009197/5238ed51-64ce-4d3f-8947-836496696e30)
![Screenshot 2024-05-31 at 11 56 28](https://github.com/strapi/strapi/assets/166009197/b114046c-f583-4dda-9426-08a3d7699a79)
![Screenshot 2024-05-31 at 12 03 42](https://github.com/strapi/strapi/assets/166009197/61e0d451-ca61-45d9-bf91-60ff6001d728)

### Code snippets

`{
  "kind": "singleType",
  "collectionName": "footer_bars",
  "info": {
    "singularName": "footer-bar",
    "pluralName": "footer-bars",
    "displayName": "FooterBar"
  },
  "options": {
    "draftAndPublish": true
  },
  "pluginOptions": {
    "versions": {
      "versioned": true
    },
    "i18n": {
      "localized": true
    }
  },
  "attributes": {
    "nav_items": {
      "type": "relation",
      "relation": "oneToMany",
      "target": "api::navbar-item.navbar-item"
    },
    "socialsLabel": {
      "pluginOptions": {
        "versions": {
          "versioned": true
        },
        "i18n": {
          "localized": true
        }
      },
      "type": "string",
      "required": true
    }
  }
}
`

**Comment by Ariseni** [comment]
@derrickmehaffy same as for the other bug
this might be connected with the content versioning plugin. Since I've removed it things have improved. And new types work better. Can't be sure without more testing so please don't close this yet

**Comment by ivo-ljubicic** [comment]
@derrickmehaffy same as for the other bug
this might be connected with the content versioning plugin. Since I've removed it things have improved. And new types work better. Can't be sure without more testing so please don't close this yet

**Comment by bolg55** [comment]
There is more related misbehavior as well. As referenced in internal TID7009 

When duplicating a content entry with two or more locales set, editing the duplicated entry will overwrite the original entry. 

I was able to reproduce this:

Specific criteria:
1. The original context to be duplicated must already have at least 2 locales assigned to it
2. The field being updated is set to "This value is common to all locales"

![ScreenRecording2024-06-05at3 58 50PM-ezgif com-video-to-gif-converter](https://github.com/strapi/strapi/assets/43022248/31d0361c-189c-43ed-b94f-27073a70624f)

**Comment by joshuaballas** [comment]
Any update on the status of this issue? Our enterprise has been version locked to Strapi v4.14.5 due to this issue and our rep reached out telling us that we can upgrade to Strapi 5. We are unaware if this issue has been resolved in Strapi 5 and cannot update until then.

**Comment by ivo-ljubicic** [comment]
We've removed all the plugins and then recreated the application. Since then (and removing versioning control) it seems to work fine.
Strapi5 has versioning control included which i just tried today and that worked nicely (can't say it won't break later though since i don't have enough information)

**Comment by joshuaballas** [comment]
We don't use any plugins. We got a call with our rep scheduled to go over this in more detail because we have upwards of a thousand properties in Strapi and we've had issues with data corruption in the past when doing updates, so we are hesitant. We'll definitely be doing a full backup of our MySQL db before trying out Strapi 5.

**Comment by derrickmehaffy** [comment]
related to: https://github.com/strapi/strapi/issues/21850

**Comment by bolg55** [comment]
Internal EE TID10816 facing the same issue

**Comment by maZahaca** [comment]
It's really hard to do anything with that bug in strapi with multiple locales, as it overrides any work done by content team in other locales in one button click.

We have such behavior with:
strapi version: v5.12.1
Community Edition
node version: v22.17.0

Any plans or ideas when this might be fixed?

**Comment by Ariseni** [comment]
Agreed. The only thing i could do was to redo everything. Was pretty lucky it was a minor redo and  later it didnt happen again.

**Comment by maZahaca** [comment]
It seems that issues still persist even moving further to strapi 5. I do understand that it's not a huge priority of fixing older version 4, but the issue seems to be serious stopper of using strapi at all.

**Comment by damen-concourse** [comment]
Hey Guys,

Is there any known workaround for this issue -- other than just not using localization? It looks like this bug has been around for quite a while with no resolution.

We’re kicking off a new project using Strapi, and this is seriously making us reconsider our choice. Localization is a core requirement for us, and this bug is a showstopper.

Happy to be corrected if we’re using localization incorrectly -- but from what I can tell, we’re not the only ones running into this problem.

Really hoping to get some clarity or traction on this. Any help or updates would be appreciated.

**Comment by Ariseni** [comment]
For me, if it failed, it failed instantly. Adding languages later was fine. Can't be sure.
Never found otu the answer. They made this a duolicate of something but it's not the same, I believe.

**Comment by damen-concourse** [comment]
It’s genuinely concerning that a localization bug this significant has remained unresolved for over a year. 

I really like what I’ve seen from this CMS so far, but issues like this -- especially when left unaddressed for so long -- make it hard to take Strapi seriously for production use.

If anyone has a working localization setup, I’d appreciate hearing how you’ve managed it. In our case, localized fields are still overwriting our original English fields, just as was reported back in May 2024.

https://github.com/strapi/strapi/issues/20409#issuecomment-2150931849

**Comment by Badlapje** [comment]
I'd like to upvote this, we also have this problem and it's really annoying.

**Comment by Badlapje** [comment]
@maccomaccomaccomacco  After investigating this issue it seems this is down to a UX problem.  There are two ways of localizing things:
1. localize a field
2. localize a content type

I assumed that when you localize a content type you also localize a field.  Because otherways you don't localize it.  Seems logical, but in poit of fact it doesn't work that way.  If you localize a content type but you don't localize a field, that field gets shared across locales.  Which means that the user will think he's overwriting a different locale, when really he's not as it is a shared field.

At the very least you should make this clear in the UI, but when marking a content type as localizable, you should ask which fields should be included and which shouldn't be.  

As to the part of making this clear in the UI, there is supposedly a demarcation for this, a globe or cut globe symbol next to the field label:

<img width="949" height="85" alt="Image" src="https://github.com/user-attachments/assets/82d00fe2-22c1-49d0-9b61-c97dae6f2c24" />

However, this doesn't seem to be reliable.  In the example above, the address isn't shared across translations if i change it in one.  Even though it is marked as a common field.  The description field, which was marked as translatable, was in fact a common field.  And the country field which was marked as translatable was translatable 🤷 

So you have several bugs here:
1. a UX bug, quite serious as it'll make people not use STRAPI or get quite frustrated with it, which might push them elsewhere.  Not to mention economic consequences for some types of sites when a content editor translates content but it overwrites that for the original locale.
2. a bug in which fields are being shown as common across locales and which aren't.

I'm on version 4.latest, so i doubt you'll fix this.  But apparently this problem persists in 5.  If i were you however, I'd fix it in both.

**Comment by OcneanuVlad** [comment]
internal note: TID 2508, TID 1057, TID 3396 facing the same issue

**Comment by patiparnne** [comment]
Still facing same issue with the lastest version 5.26.0

**Comment by Adzouz** [comment]
Hi there, I'm trying to understand everything about this issue and while running some tests on my side, I figured out a few problems that I tried to address in my latest PR [#24605](https://github.com/strapi/strapi/pull/24605)
I would like to know if the issues I've been fixing are the same as yours so I created an experimental version if someone wants to try and say if it fixes their problem: 0.0.0-experimental.05c808da4ba23c7d692755e4be824a7e89c8b752
Thanks a lot for your help and patience 🙏

**Comment by Adzouz** [comment]
Hi,
I just merged the fix and it will be part of the next Wednesday release 🚀
Closing the issue now, thanks 🙏
